pkg/apis/coordination/types.go

```diff
@@ -67,3 +93,70 @@ type LeaseList struct {
 	// items is a list of schema objects.
 	Items []Lease
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LeaseCandidate defines a candidate for a lease object.
+// Candidates are created such that coordinated leader election will pick the best leader from the list of candidates.
+type LeaseCandidate struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	Spec LeaseCandidateSpec
+}
+
+// LeaseCandidateSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// LeaseName is the name of the lease for which this candidate is contending.
+	// This field is immutable.
+	LeaseName string
+	// PingTime is the last time that the server has requested the LeaseCandidate
+	// to renew. It is only done during leader election to check if any
+	// LeaseCandidates have become ineligible. When PingTime is updated, the
+	// LeaseCandidate will respond by updating RenewTime.
```

Comment:
```suggestion
	// LeaseCandidate will respond by updating RenewTime and by removing the PingTime.
```

At least that's what I read from the controller code.

Reply:
Resetting this field is actually not a good idea IMO. From `RenewTime + someAllowedDrift > PingTime` we could derive eligibility for election without holding state in the controller.

Reply:
We can also derive eligibility by looking at the state of LeaseCandidate and Lease seeing if the leader is the most optimal. I think that way is simpler?